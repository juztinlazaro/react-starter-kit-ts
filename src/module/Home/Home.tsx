import * as React from 'react';

const Home = () => {
  const [status, setStatus] = React.useState(false);
  return (
    <div>
      <h1>Home</h1>

      {status ? 'true' : 'false'}

      <button onClick={() => setStatus(!status)}>something</button>
    </div>
  );
};

export default Home;
