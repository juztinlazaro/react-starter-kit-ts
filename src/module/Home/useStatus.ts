import { useState, useCallback } from "react";

const useStatus = (value: boolean) => {
  const [status, setStatus] = useState(value);
  useCallback(() => setStatus(value), [value]);

  return { status, setStatus };
};

export default useStatus;
