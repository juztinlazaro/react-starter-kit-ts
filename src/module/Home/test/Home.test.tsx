import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Home from "../Home";
import { renderHook } from "@testing-library/react-hooks";
import useStatus from "../useStatus";

Enzyme.configure({ adapter: new Adapter() });

describe("test home", () => {
  const HomeWrapper = shallow(<Home />);

  // tslint:disable-next-line:react-hooks-nesting
  const { result } = renderHook(() => useStatus(false));

  it("should test", () => {
    expect(result.current.status).toBe(false);
    expect(typeof result.current.setStatus).toBe("function");
  });

  it("should be false", () => {
    const status = HomeWrapper.find("p");
    expect(status.text()).toEqual("false");
  });

  it("should be true", () => {
    HomeWrapper.find("button").simulate("click");
    const status = HomeWrapper.find("p");
    expect(status.text()).toEqual("true");
  });
});
