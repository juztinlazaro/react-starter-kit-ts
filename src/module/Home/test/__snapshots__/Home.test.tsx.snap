// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test element should match to snapshot 1`] = `ShallowWrapper {}`;
